request_backend_var -> request_downloader_var (#51)
scrapinghub/web-poet: docs/advanced/additional-requests.rst

    * Since we now have an HTML response, using :meth:`~.HttpResponseBody.json`
      method would raise a ``JSONDecodeError`` as a JSON document cannot be
      parsed from it.
    * The :meth:`~.HttpResponse.selector` property is an instance of
      :external:py:class:`parsel.selector.Selector`; there are also
      :meth:`~.HttpResponse.css` and :meth:`~.HttpResponse.xpath` methods.

>         * Usually there's no need to use :meth:`~.HttpResponse.selector`,
>           as the :meth:`~.HttpResponse.css` and :meth:`~.HttpResponse.xpath`
>           are available.

Reviewer: We should remove that “the”:

```suggestion
        * Usually there's no need to use :meth:`~.HttpResponse.selector`, as 
          :meth:`~.HttpResponse.css` and :meth:`~.HttpResponse.xpath` are 
          available.
```

or append “method”:

```suggestion
        * Usually there's no need to use :meth:`~.HttpResponse.selector`, as 
          the :meth:`~.HttpResponse.css` and :meth:`~.HttpResponse.xpath` 
          methods are available.
```

    * A ``GET`` request can be done via :class:`~.HttpClient`'s
      :meth:`~.HttpClient.get` method.

>     * There is no need to instantiate a :class:`~.HttpRequest` when
>       :meth:`~.HttpClient.get` is used.

Reviewer: ```suggestion
    * There is no need to instantiate :class:`~.HttpRequest` when
      :meth:`~.HttpClient.get` is used.
```

or

```suggestion
    * There is no need create an instance of :class:`~.HttpRequest` when
      :meth:`~.HttpClient.get` is used.
```

.. tip::

>     The :meth:`~.HttpClient.batch_execute` method can multiple
>     :class:`~.HttpRequest` instances. For example, it could be a mixture
>     of ``GET`` and ``POST`` requests or even

Reviewer: ```suggestion
    The :meth:`~.HttpClient.batch_execute` method can execute multiple
    :class:`~.HttpRequest` instances. For example, it could be a mixture
    of ``GET`` and ``POST`` requests or even
```

        async def to_item(self):
            ...

>     # Once this is set, the ``request_implementation`` becomes available to
>     # all instances of HttpClient, unless HttpClient is created with
>     # ``request_downloader`` argument
>     # (see #2 Dependency Injection example below).

Reviewer: ```suggestion
    # Once this is set, the ``request_implementation`` becomes available to
    # all instances of HttpClient, unless HttpClient is created with
    # the ``request_downloader`` argument (see the #2 Dependency Injection 
    # example below).
```

    web_poet.request_downloader_var.set(request_implementation)

    # Assume that it's constructed with the necessary arguments taken somewhere.
    response = web_poet.HttpResponse(...)

    page = SomePage(response=response, http_client=create_http_client())
    item = await page.to_item()

When the ``web_poet.request_downloader_var`` contextvar is set,
:class:`~.HttpClient` instances use it by default.

.. warning::

>     If no value for ``web_poet.request_downloader_var`` is set, then a
>     :class:`~.RequestDownloaderVarError` is raised. However, no exception is
>     raised if **option 2** below is used.

Reviewer: ```suggestion
    If no value for ``web_poet.request_downloader_var`` is set, then
    :class:`~.RequestDownloaderVarError` is raised. However, no exception is
    raised if **option 2** below is used.
```

2. Dependency Injection
***********************

> The framework using **web-poet** may be using libraries which doesn't

Reviewer: ```suggestion
The framework using **web-poet** may be using libraries that don't
```

    response = web_poet.HttpResponse(...)

    page = SomePage(response=response, http_client=create_http_client())
    item = await page.to_item()

From the code sample above, we can see that every time an :class:`~.HttpClient`

> is created for Page Objects needing an ``http_client``, framework must create
> :class:`~.HttpClient` with the specific **request downloader implementation**,
> using ``request_downloader`` argument.

Reviewer: ```suggestion
instance is created for Page Objects needing an ``http_client``, the framework 
must create :class:`~.HttpClient` with a framework-specific **request 
downloader implementation**, using the ``request_downloader`` argument.
```

*********

Frameworks that handle **web-poet** MUST be able to ensure that Page Objects

> having additional requests using the :class:`~.HttpClient` are able to work
> with any type of HTTP downloader implementation.

Reviewer: ```suggestion
having additional requests using :class:`~.HttpClient` are able to work with 
any type of HTTP downloader implementation.
```
